### ISSUE TITLE
THE 1.x Stuff that needs to be fixed MEGALIST

### ISSUE THREAD

**Issue by stefanpenner** [broccolijs/broccoli#332]
Broccoli 1.x is out, but the community is split. Ember-cli for example is not yet on 1.0. This issue thread is for people reporting things that are broken, plugins that don't work, or ecosystems that expect to but are unable to benefit from it.

* ember-cli needs to be upgraded
  * lots of custom fixes that need to be forward ported
  * list needs to be made

A sample app I'm working on getting working: https://github.com/stefanpenner/es6-promise

Issues:

- [x] broccoli-watchify doesn't work
   * https://github.com/stefanpenner/es6-promise/blob/97478eb6fcd628e0d679438c88bd64b5079a9122/Brocfile.js#L70
   * repo issue:  https://github.com/eploko/broccoli-watchify/pull/10
- [x] some `stew.find` doesn't work:
   * example: https://github.com/stefanpenner/es6-promise/blob/master/Brocfile.js#L77

With those two issue hacked out, the build does complete! (So thats a good sign)

Likely issues (needs testing):

- [ ] ember-browserify
- [ ] ember-cli-sass / ember-cli-eyeglass
- [x] ?????

We should be sure:

* tests pass \w 1.x (and CI tests 1.x)
* consider upgrading test stuff to use broccoli-test-helper

Most of these do seem to work, but getting tests upgraded and passing seems like an important step.

- [x] broccoli-merge-trees
- [x] broccoli-plugin
- [x] broccoli-persistent-filter
- [x] broccoli-filter (we may want to merge this with persistent-filter)
- [x] broccoli-funnel
- [x] broccoli-concat
- [x] broccoli-stew (https://github.com/stefanpenner/broccoli-stew/blob/master/package.json#L47 claims it is, but find seems to have issues) https://github.com/stefanpenner/broccoli-stew/issues/146
- [x] broccoli-test-helper

**Comment by oligriffiths** [comment]
I’m going to throw documentation into the ring. It need a LOT of love. I’ll also throw my name into the hat for helping.

**Comment by stefanpenner** [comment]
Totally agree. I would like to keep that separate of this list as it deserves its own :p

**Comment by oligriffiths** [comment]
Fair. Shall we make a new issue?

**Comment by kellyselden** [comment]
Thank you for tackling this.

**Comment by locks** [comment]
@stefanpenner I would like to standardize the various plugins a bit more, as in move the code to ES2015. Can that be contemplated here, or a separate list?

**Comment by locks** [comment]
Deprecate find? https://github.com/stefanpenner/broccoli-stew/issues/122

**Comment by cafreeman** [comment]
Is there already a list of the known blockers for moving ember-CLI to 1.x? It seems like a full accounting of the discrepancies between Ember’s fork and broccoli would go a long way toward helping prioritize other issues

**Comment by oligriffiths** [comment]
Debugability ine one oft-cited gripe with broccoli. It’s great when it works but due to the multi stage transforms, finding out what stage went wrong is a nightmare. We should investigate a way of making this easier.

**Comment by stefanpenner** [comment]
Is the debuggability issue around even with broccoli stew’s log and debug?

**Comment by oligriffiths** [comment]
IMO it’s around broccoli. The fact that to make debugging easier you need to install stew shows that out of the box it’s hard to debug. I think it should be included in the core. Also the API for debugging with three requires manually adding stuff to your build chain, surely we can find a way to do this internally.

**Comment by stefanpenner** [comment]
So conceptually stew and log help, but the ergonomics of using them is currently the limiting factor?

**Comment by stefanpenner** [comment]
(If so I would agree) just want to make sure I understand your feedback correctly

**Comment by oligriffiths** [comment]
Correct. I also think there should be a way for broccoli to track the mutations of builds over time and be able to output a log of all the transformations when something fails, basically building in log and debug into the core.

**Comment by stefanpenner** [comment]
I've cleaned up this issue, it is intended for HARD Blockers. e.g. things that are absolutely broken on 1.x that must not be. Although many things can be improved (docs/debugging etc), this is reserved for absolute blockers.

**Comment by oligriffiths** [comment]
Seems fair. 
How should we begin to tackle this? Are there items that need to be done first?

**Comment by stefanpenner** [comment]
Well, we are all on 1.0 (2.0 or 3.0) so this can be closed.
